### PR TITLE
add support for FailurePolicy in job spec

### PR DESCRIPTION
**Pull request by nilsgstrabo**
- update radix-operator ref to v1.67.0
- update to alpine 3.21
- create local models for Node, Resources and FailurePolicy instead of using radixv1 structs -> well defined openapi schema
- update tests

### CODE REVIEW

File: models/common/resources.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package common
+
+import (
+	radixv1 "github.com/equinor/radix-operator/pkg/apis/radix/v1"
+)
+
+type ResourceList map[string]string
+
+// ResourceRequirements describes the compute resource requirements.
```
Comment (satr) [inline models/common/resources.go:9]: Leftover name

File: models/common/job_component_description.go [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
 	// Resource describes the compute resource requirements.
 	//
 	// required: false
-	Resources *v1.ResourceRequirements `json:"resources,omitempty"`
+	Resources *Resources `json:"resources,omitempty"`
 	// Node defines node attributes, where container should be scheduled
 	//
 	// required: false
-	Node *v1.RadixNode `json:"node,omitempty"`
+	Node *Node `json:"node,omitempty"`
```
Comment (satr) [inline models/common/job_component_description.go:12]: Maybe this need to consistent with radix-operator type https://github.com/equinor/radix-operator/blob/6c95cb419117d8ffa948fb9802230805958d86d3/pkg/apis/radix/v1/radixapptypes.go#L434

**Review by satr — APPROVED** [review]
Approved with comments

File: models/common/job_component_description.go (hunk 2 above)
Comment (nilsgstrabo) [reply models/common/job_component_description.go:12]: What is inconsistent?

**Review by satr — APPROVED** [review]

**Review by satr — APPROVED** [review]

**Pull request merged**